### PR TITLE
Sort out optional sender and receiver

### PR DESCRIPTION
**Pull request by gaborh-da**

### CODE REVIEW

**Comment by gaborh-da** [comment]
1. What would be a good name? TransferIn, TransferOut, InternalTransfer sender receiver - or something else?
2. In Java, I had to copy/duplicate toIBANs because one of the tests need to use it and in the original class, it can be private, or should it rather be a public static one?

File: daml-model/daml/Workflow/Data.daml [hunk 1]
```diff
@@ -26,3 +33,13 @@ data Leg = Leg with
...
+
+getSender : IBANs -> Optional Text
+getSender (SenderOnly iban) = Some iban
+getSender SenderAndReceiver{sender=iban} = Some iban
+getSender _ = None
+
+getReceiver : IBANs -> Optional Text
+getReceiver (ReceiverOnly iban) = Some iban
+getReceiver SenderAndReceiver{receiver=iban} = Some iban
+getReceiver _ = None
```
Comment (tamaskalcza-da) [inline daml-model/daml/Workflow/Data.daml:45]: What do you think of this?
```suggestion
senderOf : SettlementStep -> Optional Text
senderOf (SettlementStep {ibans = SenderOnly iban, .. }) = Some iban
senderOf (SettlementStep {ibans = SenderAndReceiver{sender=iban}, .. }) = Some iban
senderOf _ = None

receiverOf : SettlementStep -> Optional Text
receiverOf (SettlementStep {ibans = ReceiverOnly iban, .. }) = Some iban
receiverOf (SettlementStep {ibans = SenderAndReceiver{receiver=iban}, .. }) = Some iban
receiverOf _ = None
```

File: integration-adapter/src/main/java/com/rln/conversion/kafka2daml/EnrichedPacs008SwiftToDamlTranslation.java [hunk 2]
```diff
@@ -76,6 +79,23 @@ private List<Tuple2<String, SettlementStep>> translateBics(String initiatorParty
...
+            if (hasReceiver) {
+                return new SenderAndReceiver(step.getSender(), step.getReceiver());
+            } else {
+                return new SenderOnly(step.getSender());
+            }
+        } else if (hasReceiver) {
+            return new ReceiverOnly(step.getReceiver());
+        } else {
+            throw new RuntimeException(String.format("Step missing both sender and receiver: %s", step));
+        }
```
Comment (tamaskalcza-da) [inline integration-adapter/src/main/java/com/rln/conversion/kafka2daml/EnrichedPacs008SwiftToDamlTranslation.java:99]:
```suggestion
        var sender = step.getSender();
        var receiver = step.getReceiver();
        if (sender == null && receiver == null) {
            throw new RuntimeException(String.format("Step missing both sender and receiver: %s", step));
        }
        if (sender == null) return new ReceiverOnly(receiver);
        if (receiver == null) return new SenderOnly(sender);
        return new SenderAndReceiver(sender, receiver);
```

File: integration-adapter/src/test/java/com/rln/CommonBaseTest.java [hunk 3]
```diff
@@ -225,4 +230,20 @@ protected void eventually(Runnable code) throws InterruptedException {
             }
         }
     }
+
+    private static IBANs toIBANs(Step step) {
```
Comment (tamaskalcza-da) [inline integration-adapter/src/test/java/com/rln/CommonBaseTest.java:234]: Seems like we have this code already. I'd move that code to either a commonly accessibly utility function, or maybe into `Step`, although the latter is a bit questionable.

Comment (gaborh-da) [reply integration-adapter/src/test/java/com/rln/CommonBaseTest.java]: I introduced a Utility.java

**Review by tamaskalcza-da — APPROVED** [review]

**Review by tamaskalcza-da — APPROVED** [review]

**Pull request merged**